src/debugger/debugAdapterFactory.ts

```diff
@@ -0,0 +1,98 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2021-2023 the VSCode Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VSCode Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as vscode from "vscode";
+import * as fs from "fs";
+import { WorkspaceContext } from "../WorkspaceContext";
+import { SwiftToolchain } from "../toolchain/toolchain";
+
+export function registerLLDBDebugAdapter(workspaceContext: WorkspaceContext): vscode.Disposable {
+    class LLDBDebugAdapterExecutableFactory implements vscode.DebugAdapterDescriptorFactory {
+        // The following use of a DebugAdapter factory shows how to control what debug adapter executable is used.
+        // Since the code implements the default behavior, it is absolutely not neccessary and we show it here only for educational purpose.
+
+        createDebugAdapterDescriptor(
+            _session: vscode.DebugSession,
+            executable: vscode.DebugAdapterExecutable | undefined
+        ): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
+            // param "executable" contains the executable optionally specified in the package.json (if any)
+
+            // use the executable specified in the package.json if it exists or determine it based on some other information (e.g. the session)
+            if (!executable) {
+                executable = new vscode.DebugAdapterExecutable(
+                    workspaceContext.toolchain.getToolchainExecutable("lldb-vscode"),
+                    [],
+                    {}
+                );
+            }
+
+            // make VS Code launch the DA executable
+            return executable;
+        }
+    }
+
+    const debugAdpaterFactory = vscode.debug.registerDebugAdapterDescriptorFactory(
+        "swift-lldb",
+        new LLDBDebugAdapterExecutableFactory()
+    );
+    const debugConfigProvider = vscode.debug.registerDebugConfigurationProvider(
+        "swift-lldb",
+        new LLDBDebugConfigurationProvider()
+    );
+    return {
+        dispose: () => {
+            debugConfigProvider.dispose();
+            debugAdpaterFactory.dispose();
+        },
+    };
+}
+
+async function isFileExists(path: string): Promise<boolean> {
```

Comment:
This is a weird name "is file exists".

Reply:
Haha copy and paste from Stackoverflow. I'll give it a better name.